tracer/src/Datadog.Trace/Configuration/ConfigurationSources/Telemetry/ConfigurationBuilder.cs

```diff
@@ -140,6 +219,74 @@ public string AsString(string defaultValue, Func<string, bool>? validator)
             return defaultValue.Result!;
         }
 
+        [return: NotNullIfNotNull(nameof(getDefaultValue))]
+        public T? GetAs<T>(Func<DefaultResult<T>>? getDefaultValue, Func<T, bool>? validator, Func<string, ParsingResult<T>> converter, string openTelemetryKey, Func<string, ParsingResult<T>> openTelemetryConverter)
+        {
+            var result = Source.GetAs<T>(Key, Telemetry, converter, validator, recordValue: true)
+                      ?? (FallbackKey1 is null ? null : Source.GetAs<T>(FallbackKey1, Telemetry, converter, validator, recordValue: true))
+                      ?? (FallbackKey2 is null ? null : Source.GetAs<T>(FallbackKey2, Telemetry, converter, validator, recordValue: true))
+                      ?? (FallbackKey3 is null ? null : Source.GetAs<T>(FallbackKey3, Telemetry, converter, validator, recordValue: true));
+
+            T? returnValue = default;
+            bool datadogConfigurationIsPresent = false;
+            if (result is { Result: { } value, IsValid: { } resultIsValid })
+            {
+                datadogConfigurationIsPresent = true;
+                if (resultIsValid)
+                {
+                    returnValue = value;
+                }
+            }
+            else
+            {
+                datadogConfigurationIsPresent = Source.IsPresent(Key)
+                                             || (FallbackKey1 is null ? false : Source.IsPresent(FallbackKey1))
+                                             || (FallbackKey2 is null ? false : Source.IsPresent(FallbackKey2))
+                                             || (FallbackKey3 is null ? false : Source.IsPresent(FallbackKey3));
+            }
+
+            // If there's a Datadog configuration present, check if a corresponding OpenTelemetry key is present so we can log the conflicting keys
+            if (datadogConfigurationIsPresent && Source.IsPresent(openTelemetryKey))
+            {
+                Log.Warning(
+                    "Both Datadog configuration {DatadogConfiguration} and OpenTelemetry configuration {OpenTelemetryConfiguration} are set. The Datadog configuration will be used.",
+                    Key,
+                    openTelemetryKey);
+                OpenTelemetryHelpers.GetConfigurationMetricTags(openTelemetryKey, out var openTelemetryConfig, out var datadogConfig);
+                TelemetryFactory.Metrics.RecordCountOpenTelemetryConfigHiddenByDatadogConfig(datadogConfig, openTelemetryConfig);
+            }
+            else if (Source.IsPresent(openTelemetryKey))
```

Comment:
I'm wondering about a different approach to trying to do this all in one... Instead of trying to do everything in the configuration builder, could/should we move the decision making + telemetry recording into the `TracerSettings` object itself? 

FWIW, I think we _have_ to do this, because I don't think we can safely record the telemetry here 🙁

Reply:
In e8fc0b590dbb733caa248882d8c6215cdd9a4a57 I've removed the telemetry calls from this class, so this should reduce risk. I think that may be the right solution to enable telemetry, but for now this fulfills the requirements as covered in the unit tests so I wonder if we can keep this implementation for the initial PR

Reply:
In e8fc0b590dbb733caa248882d8c6215cdd9a4a57 I've removed the telemetry calls from this class, so this should reduce risk. I think that may be the right solution to enable telemetry, but for now this fulfills the requirements as covered in the unit tests so I wonder if we can keep this implementation for the initial PR

Reply:
Yeah, that's fine by me - I'm also happy to look at it, this infra stuff probably falls under lang-platform tbh, so I'd like to improve this later 🙂